### PR TITLE
PR: Avoid KeyboardInterrupt to raise an error in waitingspinner

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes: #5114

### CODE REVIEW

File: spyder/app/mainwindow.py [hunk 1]
```diff
@@ -288,6 +288,7 @@ def signal_handler(signum, frame=None):
                 pass
         else:
             signal.signal(signal.SIGTERM, signal_handler)
+            signal.signal(signal.SIGINT, signal_handler)
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Why is this here?

Comment (rlaverde) [reply spyder/app/mainwindow.py]: https://github.com/spyder-ide/spyder/issues/5114#issuecomment-327024883

It make possible to kill spyder using `Ctrl+C`, without this change It will still raise the report error message box when pressing `Ctrl+C`

Comment (ccordoba12) [reply spyder/app/mainwindow.py]: Please remove this. We need to discuss it in another PR.

Comment (goanpeca) [reply spyder/app/mainwindow.py]: @ccordoba12 why? I think it is important for people to be able to quit spyder on the terminal using `Ctrl+C`

Comment (ccordoba12) [reply spyder/app/mainwindow.py]: When Spyder freezes for whatever reason, it's very useful to press `Ctrl+C` in the terminal and get the traceback of where it was freezed.

Comment (rlaverde) [reply spyder/app/mainwindow.py]: But, the users will continue reporting false positives (now random), of when they tried to close spyder with `Ctrl+C`

Comment (ccordoba12) [reply spyder/app/mainwindow.py]: But that only happens when users press `Ctrl+C` in a terminal, and I doubt there are many of them.

In any case, I think this needs more discussion and is not related to this PR. So please remove it and open a new PR with change only.

**Comment by rlaverde** [comment]
@ccordoba12 I move that changes to a new PR #5159 and let only waiting spinner changes in this PR

**Comment by ccordoba12** [comment]
Great! Thanks @rlaverde!
